lib/ABI/ModelAlignment.cpp

```diff
@@ -0,0 +1,171 @@
+/// \file ModelAlignment.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ABI/ModelAlignment.h"
+#include "revng/ABI/RuntimeTraitAccessors.h"
+#include "revng/Model/Binary.h"
+
+namespace abi {
+
+static RecursiveCoroutine<std::optional<uint64_t>>
+naturalAlignmentImpl(model::VerifyHelper &VH,
+                     const model::Type &Type,
+                     model::ABI::Values ABI);
+static RecursiveCoroutine<std::optional<uint64_t>>
+naturalAlignmentImpl(model::VerifyHelper &VH,
+                     const model::QualifiedType &Type,
+                     model::ABI::Values ABI);
+
+template<typename FieldType>
+static RecursiveCoroutine<std::optional<uint64_t>>
+aggregateAlignmentImpl(model::VerifyHelper &VH,
+                       const SortedVector<FieldType> Fields,
+                       model::ABI::Values ABI) {
+  uint64_t Alignment = 0;
+
+  for (const auto &Field : Fields) {
+    if (auto A = rc_recur naturalAlignmentImpl(VH, Field.Type, ABI))
+      Alignment = std::max(Alignment, *A);
+    else
+      rc_return std::nullopt;
+  }
+
+  rc_return Alignment;
+}
+
+static RecursiveCoroutine<std::optional<uint64_t>>
+naturalAlignmentImpl(model::VerifyHelper &VH,
+                     const model::Type &Type,
+                     model::ABI::Values ABI) {
+  auto MaybeAlignment = VH.size(&Type);
```

Comment:
It seems like you're reusing `VerifyHelper`'s size to represent alignment. You can extend `VerifyHelper` to handle alignment as an additional thing. Why abuse an existing feature?

Reply:
Done.